src/dotnet/CarbonAware.WebApi/Controllers/SciScoreController.cs

```diff
@@ -1,45 +1,125 @@
+            var error = new CarbonAwareWebApiError() { Message = "Location is required" };
+            return BadRequest(error);
         }
 
-        if (String.IsNullOrEmpty(calculation.Duration))
+        if (String.IsNullOrEmpty(input.TimeInterval))
         {
-            return BadRequest("Duration is required");
+            var error = new CarbonAwareWebApiError() { Message = "TimeInterval is required" };
+            return BadRequest(error);
         }
 
         SciScore score = new SciScore
+        {
+            SciScoreValue = 100.0,
+            EnergyValue = 1.0,
+            MarginalCarbonEmissionsValue = 100.0,
+            EmbodiedEmissionsValue = 0.0,
+            FunctionalUnitValue = 1
+        };
+
+        return await Task.Run(() => Ok(score));
+    }
+
+    [HttpPost("marginal-carbon-intensity")]
+    [Consumes(MediaTypeNames.Application.Json)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> GetCarbonIntensityAsync(SciScoreInput input)
+    {
+        _logger.LogInformation(" calling to aggregator to ");
+        if (input.Location == null)
+        {
+            var error = new CarbonAwareWebApiError() { Message = "location is required" };
+            return BadRequest(error);
+        }
+
+        if (String.IsNullOrEmpty(input.TimeInterval))
+        {
+            var error = new CarbonAwareWebApiError() { Message = "timeInterval is required" };
+            return BadRequest(error);
+        }
+
+        try
+        {
+            var location = this.GetLocation(input.Location);
+            var carbonIntensity = await _aggregator.CalculateAverageCarbonIntensityAsync(location, input.TimeInterval);
+            SciScore score = new SciScore
             {
-                SciScoreValue = 100.0f,
-                EnergyValue = 1.0f,
-                MarginalCarbonEmissionsValue = 100.0f,
-                EmbodiedEmissionsValue = 0.0f,
-                FunctionalUnitValue = 1
+                MarginalCarbonEmissionsValue = carbonIntensity,
             };
 
-        return await Task.Run(() => Ok(score));
+            return Ok(score);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError("Exception occured during marginal calculation execution", ex);
+            var error = new CarbonAwareWebApiError() { Message = ex.ToString() };
+            return BadRequest(error);
+        }
+
     }
-}
+
+    private Location GetLocation(LocationInput locationInput)
+    {
+        LocationType locationType;
+        CloudProvider cloudProvider;
+
+        if (!Enum.TryParse<LocationType>(locationInput.LocationType, true, out locationType))
+        {
+            throw new ArgumentException($"locationType '{locationInput.LocationType}' is invalid");
+        }
+
+        Enum.TryParse<CloudProvider>(locationInput.CloudProvider, true, out cloudProvider);
```

Comment:
And ditto here. Is this possible? I.e., is it possible to create a locationInput with an invalid CloudProvider since it has to be an enum?